src/common.jl

```diff
@@ -0,0 +1,161 @@
+    saveat=[],callback=()->nothing,timeseries_errors=true,dense_errors=false,
+    kwargs...)
+  tspan = prob.tspan
+
+  if tspan[end]-tspan[1]<tType(0)
+    error("final time must be greater than starting time. Aborting.")
+  end
+  atomloaded = isdefined(Main,:Atom)
+  o = KW(kwargs)
+  t = tspan[1]
+  u0 = prob.u0
+  o[:T] = tspan[end]
+
+  if typeof(u0) <: Number
+    u = [u0]
+  else
+    u = deepcopy(u0)
+  end
+
+  sizeu = size(u)
+
+  opts = buildOptions(o,ODEJL_OPTION_LIST,ODEJL_ALIASES,ODEJL_ALIASES_REVERSED)
+
+  if !isinplace && typeof(u)<:AbstractArray
+    f! = (t,u,du) -> (du[:] = prob.f(t,u))
+  else
+    f! = prob.f
+  end
+  ode  = ODE.ExplicitODE(t,u,f!)
+  # adaptive==true ? FoA=:adaptive : FoA=:fixed #Currently limied to only adaptive
+  FoA = :adaptive
+  if typeof(alg) <: rk23
+    solver = ODE.RKIntegrator{FoA,:rk23}
+  elseif typeof(alg) <: rk45
+    solver = ODE.RKIntegrator{FoA,:dopri5}
+  elseif typeof(alg) <: feh78
+    solver = ODE.RKIntegrator{FoA,:feh78}
+  elseif typeof(alg) <: ModifiedRosenbrock
+    solver = ODE.ModifiedRosenbrockIntegrator
+  elseif typeof(alg) <: feuler
+    solver = ODE.RKIntegratorFixed{:feuler}
+  elseif typeof(alg) <: midpoint
+    solver = ODE.RKIntegratorFixed{:midpoint}
+  elseif typeof(alg) <: heun
+    solver = ODE.RKIntegratorFixed{:heun}
+  elseif typeof(alg) <: rk4
+    solver = ODE.RKIntegratorFixed{:rk4}
+  elseif typeof(alg) <: feh45
+    solver = ODE.RKIntegrator{FoA,:rk45}
+  end
+  out = ODE.solve(ode;solver=solver,opts...)
+  timeseries = out.y
+  ts = out.t
+  ks = out.dy
+  if length(out.y[1])==1
+    tmp = Vector{eltype(out.y[1])}(length(out.y))
+    tmp_dy = Vector{eltype(out.dy[1])}(length(out.dy))
+    for i in 1:length(out.y)
+      tmp[i] = out.y[i][1]
+      tmp_dy[i] = out.dy[i][1]
+    end
+    timeseries = tmp
+    ks = tmp_dy
+  end
+
+  saveat_idxs = find((x)->x∈saveat,ts)
+  t_nosaveat = view(ts,symdiff(1:length(ts),saveat_idxs))
+  u_nosaveat = view(timeseries,symdiff(1:length(ts),saveat_idxs))
+
+  if dense
+    interp = (tvals) -> common_interpolation(tvals,t_nosaveat,u_nosaveat,ks,alg,f!)
+  else
+    interp = (tvals) -> nothing
+  end
+
+  build_solution(prob,alg,ts,timeseries,
+                    dense=dense,k=ks,interp=interp,
+                    timeseries_errors = timeseries_errors,
+                    dense_errors = dense_errors)
+end
+
+const ODEJL_OPTION_LIST = Set([:tout,:tstop,:reltol,:abstol,:minstep,:maxstep,:initstep,:norm,:maxiters,:isoutofdomain])
+const ODEJL_ALIASES = Dict{Symbol,Symbol}(:minstep=>:dtmin,:maxstep=>:dtmax,:initstep=>:dt,:tstop=>:T,:maxiters=>:maxiters)
+const ODEJL_ALIASES_REVERSED = Dict{Symbol,Symbol}([(v,k) for (k,v) in ODEJL_ALIASES])
+
+function buildOptions(o,optionlist,aliases,aliases_reversed)
+  dict1 = Dict{Symbol,Any}([Pair(k,o[k]) for k in (keys(o) ∩ optionlist)])
+  dict2 = Dict([Pair(aliases_reversed[k],o[k]) for k in (keys(o) ∩ values(aliases))])
+  merge(dict1,dict2)
+end
+
+"""
```

Comment:
I assumed that sooner rather than later you might be replacing it with the interpolation functions that you already have internally (so that way it could also be specialized to the method), but wanted to make sure that the PR at least had the functionality it had before this code was moved.